pkg/components/prometheus-operator/component_conversion_test.go

```diff
@@ -0,0 +1,137 @@
+package prometheus //nolint:testpackage
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	yamlserializer "k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/client-go/util/jsonpath"
+
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+)
+
+func renderManifests(t *testing.T, configHCL string) map[string]string {
+	name := "prometheus-operator"
+
+	component := newComponent()
+
+	body, diagnostics := util.GetComponentBody(configHCL, name)
+	if diagnostics.HasErrors() {
+		t.Fatalf("Getting component body: %v", diagnostics.Errs())
+	}
+
+	diagnostics = component.LoadConfig(body, &hcl.EvalContext{})
+	if diagnostics.HasErrors() {
+		t.Fatalf("Valid config should not return an error, got: %v", diagnostics)
+	}
+
+	ret, err := component.RenderManifests()
+	if err != nil {
+		t.Fatalf("Rendering manifests with valid config should succeed, got: %s", err)
+	}
+
+	return ret
+}
+
+//nolint:funlen
+func TestConversion(t *testing.T) {
+	testCases := []struct {
+		Name                   string
+		InputConfig            string
+		ExpectedConfigFileName string
+		Expected               reflect.Value
```

Comment:
This is to make the test code generic. In a yaml file when you provide a json path, the output of json path could be string, boolean, number, list, map, etc. To keep the test code generic and to make it easier to to add more tests, here I am using generic type. 

The test code does heavy lifting of converting to correct type. If the test code does not support the concrete type yet, we(the test author) should add support for it. Keeping this API/test code like this because first it is not a user facing change so even if things fail we can fix it and second since there is plan to make the test code into helper function we have generic way of doing this. Now in the table/array of tests we can add more types to test without making change to the test struct.

Reply:
As I mentioned in #964 already, I'd also make it a string.

Reply:
I added my opinion here: https://github.com/kinvolk/lokomotive/pull/964#discussion_r500420376